File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    public List<List<Integer>> res= new ArrayList<>();

    public List<List<Integer>> levelOrder(TreeNode root) {
        order(root,0);

        return res;
    }
    public void order(TreeNode root, int level)
    {
        if(root==null) return;

        if(res.size()==level){
            res.add(new ArrayList(Arrays.asList(root.val)));
        }
        else
        {
            res.get(level).add(root.val);
        }
        order(root.left,level+1);
        order(root.right,level+1);
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxheight(self,root: Optional[TreeNode]) ->int:

        if root == None:
            return 0

        if root.left == None and root.right == None:
            return 1
        
        return 1+max(self.maxheight(root.right),self.maxheight(root.left))
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:

        if root == None:
            return 0
        
        return max((self.maxheight(root.right)+self.maxheight(root.left)), self.diameterOfBinaryTree(root.left), self.diameterOfBinaryTree(root.right))

        

        